msteams-platform/toolkit/add-single-sign-on.md

```diff
@@ -0,0 +1,346 @@
+
+   ```
+
+3. Register your command in the Teams app manifest. Open `templates/appPackage/manifest.template.json`, and add following lines under `command` in `commandLists` of your bot:
+
+   ```JSON
+
+   {
+       "title": "photo",
+       "description": "Show user photo using Single Sign On feature"
+   }
+
+   ```
+</details>
+<br>
+
+## Debug your application
+
+Press F5 to debug your application. Teams Toolkit uses the Azure AD manifest file to register an Azure AD application for SSO. For Teams Toolkit local debug functionalities, see [Debug your Teams app locally](debug-local.md).
+
+## Customize Azure AD applications
+
+The [Azure AD app manifest](/azure/active-directory/develop/reference-app-manifest) allows you to customize various aspects of application registration. You can update the manifest as needed. If you need to include additional API permissions to access your desired APIs, see [API permissions to access your desired APIs](https://github.com/OfficeDev/TeamsFx/wiki/#customize-aad-manifest-template).
+To view your Azure AD application in Azure Portal, see [View Azure AD application in Azure portal](https://github.com/OfficeDev/TeamsFx/wiki/Manage-AAD-application-in-Teams-Toolkit#How-to-view-the-AAD-app-on-the-Azure-portal). 
+
+## SSO authentication concepts
+
+The following concepts helps you for SSO authentication:
+
+### Working of SSO in Teams
+
+Single sign-on (SSO) authentication in Microsoft Azure Active Directory (Azure AD) silently refreshes the authentication token to minimize the number of times users need to enter their sign-in credentials. If users agree to use your app, they don't have to provide consent again on another device as they're signed in automatically.
+
+Teams tabs and bots have similar flow for SSO support, for more information, see:
+
+1. [Single sign-on (SSO) authentication in Tabs](../tabs/how-to/authentication/auth-aad-sso.md)
+2. [Single sign-on (SSO) authentication in Bots](../bots/how-to/authentication/auth-aad-sso-bots.md)
+
+### Simplified SSO with TeamsFx
+
+TeamsFx helps to reduce the developer tasks by using Teams SSO and accessing cloud resources down to single line statements with zero configuration.
+
+With TeamsFx SDK, you can write user authentication code in a simplified way using Credentials:
+
+1. User identity in browser environment: `TeamsUserCredential` represents Teams current user's identity.
+2. User identity in Node.js environment: `OnBehalfOfUserCredentail` uses On-Behalf-Of flow and Teams SSO token.
+3. Application Identity in Node.js environment: `AppCredential` represents the application identity.
+
+For more information about TeamsFx SDK, see:
+
+* [TeamsFx SDK](TeamsFx-SDK.md) or [API reference](/javascript/api/@microsoft/teamsfx/?view=msteams-client-js-latest&preserve-view=true)
+* [Microsoft Teams Framework (TeamsFx) Sample Gallery](https://github.com/OfficeDev/TeamsFx-Samples/tree/v2)
+* [TeamsFx-Samples](https://github.com/OfficeDev/TeamsFx-Samples/tree/v2)
```

Comment:
These two links seem to be the same thing, we just need to keep one.

Reply:
Updated as mentioned.